util/logtools/encounter_tools.ts

```diff
@@ -384,10 +388,10 @@ class TLFuncs {
     const dateStr = TLFuncs.dayFromDate(fightOrZone.startTime).replace(/-/g, '');
     const timeStr = TLFuncs.timeFromDate(fightOrZone.startTime).replace(/:/g, '');
     const duration = TLFuncs.durationFromDates(fightOrZone.startTime, fightOrZone.endTime);
-    let seal;
+    let seal: string | undefined;
     if ('sealName' in fightOrZone)
       seal = fightOrZone['sealName'];
-    if (seal)
+    if (seal !== undefined && seal.length > 0)
```

Comment:
The `sealName` comes from the match groups on the `will be sealed` log line.  I think we don't need to treat empty as a special case here.

```suggestion
    if (seal !== undefined)
```